Lib/inspect.py

```diff
@@ -970,10 +970,12 @@ def findsource(object):
 
     if isclass(object):
         try:
-            firstlineno = vars(object)['__firstlineno__']
+            lnum = vars(object)['__firstlineno__'] - 1
         except (TypeError, KeyError):
             raise OSError('source code not available')
-        return lines, firstlineno - 1
+        if lnum >= len(lines):
```

Comment:
```suggestion
        if lnum >= len(lines) or lnum < 0:
```

Just in case someone set `__firstlineno__` to 0 by accident.

Don't forget the tests with `__firstlineno__ = 0`. And maybe, you should check if `__firstlineno__` is an integer (not a float for instance)? While `co_firstlineno` is checked against non-integer values, setting `__firstlineno__` does not involve any type checks.

Reply:
This is the copy of the check for functions.

In other place, there is a check for `lnum > 0` after calling `findsource()`, so this case is already handled.

Also, it is possible to get `lnum >= len(lines)` by accident (what we see in this issue), but to get non-integer or non-positive `lnum` you should intentionally set the wrong value. Garbage in -- garbage out.